Tests/NIOTests/ByteBufferTest.swift

```diff
@@ -1704,6 +1704,37 @@ class ByteBufferTest: XCTestCase {
         XCTAssertEqual(0xcc, slice.getInteger(at: 0, as: UInt8.self))
         XCTAssertEqual(0xdd, slice.getInteger(at: slice.writerIndex - 1, as: UInt8.self))
     }
+    
+    func testSliceOnSliceAfterHitting16MBMark() {
+        // This test ensures that a slice will get a new backing storage if its start is more than
+        // 16MiB after the originating backing storage.
+        
+        // create a buffer with (16MiB - 1byte) + 2 bytes = 16MiB + 1 byte
+        let inputBufferLength = Int(_UInt24.max) + 2
+        var inputBuffer = ByteBuffer(repeating: 1, count: inputBufferLength)
```

Comment:
Let's prepare the buffer with a different byte pattern so that we can prove at the end that we read the correct one. For example:
- 8 `1`s (which will be read in the `readInteger`)
- 16 MB - 6 (?) `2`s (which will be read in the `readSlice`)
- 1 `3` (which will be the leftovers aka the "final slice")